CryptoTechReminderSystem.Test/UseCase/GetLateDevelopersTests.cs

```diff
@@ -88,30 +101,98 @@ public void CanGetHarvestDevelopers()
         public void CanGetSlackDevelopers()
         {
             var harvestGatewaySpy = new HarvestGatewaySpy();
-            var slackGatewaySpy = new SlackGatewaySpy();            
-            var getDevelopers = new GetLateDevelopers(slackGatewaySpy, harvestGatewaySpy, harvestGatewaySpy);
+            var slackGatewaySpy = new SlackGatewaySpy();
+            var clock = new ClockStub(
+                new DateTimeOffset(
+                    new DateTime(2019, 03, 01, 10, 30, 0)
+                )
+            );
+            
+            var getDevelopers = new GetLateDevelopers(slackGatewaySpy, harvestGatewaySpy, harvestGatewaySpy, clock);
             
             getDevelopers.Execute();
 
             slackGatewaySpy.IsRetrieveDevelopersCalled.Should().BeTrue();
         }
-        
-        [Test]
-        public void CanGetTimesheets()
+
+        [TestFixture]
+        public class CanGetTimeSheets
         {
-            var harvestGatewaySpy = new HarvestGatewaySpy();
-            var slackGatewayStub = new SlackGatewayStub();
+            [Test]
+            public void CanRetrieveTimeSheets()
+            {
+                var harvestGatewaySpy = new HarvestGatewaySpy();
+                var slackGatewayStub = new SlackGatewayStub();
+                var clock = new ClockStub(
+                    new DateTimeOffset(
+                        new DateTime(2019, 03, 01, 10, 30, 0)
+                    )
+                );
             
-            var getDevelopers = new GetLateDevelopers(slackGatewayStub, harvestGatewaySpy, harvestGatewaySpy);
+                var getDevelopers = new GetLateDevelopers(slackGatewayStub, harvestGatewaySpy, harvestGatewaySpy, clock);
             
-            getDevelopers.Execute();
+                getDevelopers.Execute();
 
-            harvestGatewaySpy.IsRetrieveTimeSheetsCalled.Should().BeTrue();
+                harvestGatewaySpy.IsRetrieveTimeSheetsCalled.Should().BeTrue();
+            }
+            
+            [Test]
+            [TestCase(19)]
+            [TestCase(18)]
+            [TestCase(17)]
+            [TestCase(16)]
+            [TestCase(15)]
+            public void CanRetrieveTimeSheetsWithStartingDate(int day)
+            {
+                var harvestGatewaySpy = new HarvestGatewaySpy();
+                var slackGatewayStub = new SlackGatewayStub();
+                var clock = new ClockStub(
+                    new DateTimeOffset(
+                        new DateTime(2019, 04, day)
+                    )
+                );
```

Comment:
also extract to setup

Reply:
Resolved in https://github.com/chubberlisk/crypto_tech/pull/23/commits/6be324ccbeffbc71f70eebe10457548fb15896dc.